synapseclient/__main__.py

```diff
@@ -378,6 +379,46 @@ def storeTable(args, syn):
     syn.logger.info('{"tableId": "%s"}', table_ent.tableId)
 
 
+def config(args, syn):
+    """Create/modify a synapse configuration file"""
+    user = input("Username:")
+    secret_prompt = "Auth token:"
+    passwd = None
+    while not passwd:
+        # if the terminal is not a tty, we are unable to read from standard input
+        # For git bash using python getpass
+        # https://stackoverflow.com/questions/49858821/python-getpass-doesnt-work-on-windows-git-bash-mingw64
+        if not sys.stdin.isatty():
+            raise SynapseAuthenticationError(
+                "No password, key, or token was provided and unable to read from standard input")
+        else:
+            passwd = getpass.getpass(secret_prompt)
+    # Since we don't split up credential configuration file, it is simply too
+    # hacky to try to edit the file minimally in place.  Therefore, I will
+    # copy the old configuration file into a .synapseConfig.backup and write a
+    # new .synaspeConfig file.
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    # Read in configuration
+    cur_config_path = os.path.dirname(os.path.realpath(args.configPath))
+    # Make a copy of the existing config if it exists
+    if os.path.exists(args.configPath):
+        shutil.copyfile(
+            args.configPath,
+            os.path.join(cur_config_path, f"{args.configPath}.backup")
+        )
+
+    with open(os.path.join(script_dir, ".synapseConfig"), "r") as config_o:
```

Comment:
Should there be some sort of warning that says the old config file will be copied into a new file?  If there are existing configurations, they need to be copied into the newly created config file.